src/Language/PureScript/Docs/AsHtml.hs

```diff
@@ -215,6 +215,33 @@ codeAsHtml r = outputWith elemAsHtml
             linkToDecl ns target mn (withClass class_ (text name))
         NoLink ->
           text name
+    Role role ->
+      case role of
+        "nominal" -> renderRole describeNominal "decl__role_nominal"
+        "phantom" -> renderRole describePhantom "decl__role_phantom"
+
+        -- representational is intentionally not rendered
+        _ -> toHtml ("" :: Text)
```

Comment:
Should I add an internal error here in case we get something other than `representational`?

Reply:
I've decided to add this error.